Sources/SwiftDocC/Semantics/Metadata/Availability.swift

```diff
@@ -0,0 +1,138 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2022 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+import Markdown
+
+/// A directive that sets the platform availability information for a documentation page.
+///
+/// `@Available` is analagous to the `@available` attribute in Swift: It allows you to specify a
+/// platform version that the page relates to. To specify a platform and version, list the platform
+/// name and use the `introduced` argument:
+///
+/// ```markdown
+/// @Available(macOS, introduced: "12.0")
+/// ```
+///
+/// The available platforms are `macOS`, `iOS`, `watchOS`, and `tvOS`.
+///
+/// This directive is available on both articles and documentation extension files. In extension
+/// files, the information overrides any information from the symbol itself.
+///
+/// This directive is only valid within a ``Metadata`` directive:
+///
+/// ```markdown
+/// @Metadata {
+///     @Available(macOS, introduced: "12.0")
+/// }
+/// ```
+public final class MetadataAvailability: Semantic, AutomaticDirectiveConvertible {
```

Comment:
Is there a reason to not just call this `Availability`? If so – I personally would lean towards `AvailabilityMetadata`. `MetadataAvailability` makes me think this affects the availability of the metadata.

Reply:
I feel like the term "availability" is pretty overloaded already in the Swift-DocC codebase; there's already a `SymbolGraph.Symbol.Availability` in SymbolKit which could collide in the wrong situation, but SymbolKit also has `AvailabilityItem` docc itself adds `AvailabilityRenderItem` as well. `AvailabilityMetadata` seems like a fine enough rename; i could add that if you think the existing name is too confusing.

Reply:
I think i want to keep the name, since my reasoning behind it is that it's "availability from the metadata", as opposed to "availability from the symbol" (`SymbolGraph.Symbol.Availability`). I would have nested the type declaration under `Metadata`, but the other Metadata directive types are also standalone so i decided to make it fit in with the others.

Reply:
> I would have nested the type declaration under Metadata, but the other Metadata directive types are also standalone so i decided to make it fit in with the others.

I agree with your initial thought here. My preference would be to nest this as an extension under `Metadata` – that fits in well with how the `Row`/`Column` and `TabNavigator`/`Tab` directives are structured today so we have some precedent at least and we could do the same with other commonly-named Metadata directives moving forward. (Eventually I'd like to split out directives into their own library so we don't run into these kind of collisions.) What do you think? My thinking is that it's more important to maintain the one-to-one relationship of container-class-API-name to directive-name relationship we currently have for the vast majority of directives.

Reply:
> I think i want to keep the name, since my reasoning behind it is that it's "availability from the metadata", as opposed to "availability from the symbol" (SymbolGraph.Symbol.Availability). 

My objection to this is that I think a reasonable read of both of these is "availability **for** the metadata" and "availability **for** the symbol".

Reply:
I think nesting the type under Metadata is the right move. I'll make the change.